### DISCUSSION TITLE
backslash breaks strings

### DISCUSSION THREAD

**Discussion by C4PTRAMPAGE** [cc-tweaked/CC-Tweaked#1235] [Q&A] [answered]
i want to have a spinning slash at the end of some text but putting in a backslash breaks the string, how do i put a backslash in a string without breaking it?

xy = term.setCursorPos
wr = term.write
clr = term.clear

graphic = function()
  xy(17,1)
  if graphicState == 0 then
    wr("/")
  elseif graphicState == 1 then
    wr("-")
  elseif graphicState == 2 then
    wr("\")
  elseif graphicState == 3 then
    wr("|")
  elseif graphicState == 4 then
    wr("/")
  elseif graphicState == 5 then
    wr("-")
  elseif graphicState == 6 then
    wr("\")
  elseif graphicState == 7 then
    wr("|")
  end
  graphicState = graphicState + 1
  if graphicState >= 7 then
    graphicState = 0
  end
end

graphicState = 0
clr()
xy(1,1)
wr("Systems Computer")
while true do
  graphic()
  sleep(1)
end

**Comment by Lupus590** [comment, marked as answer]
Backslash is the escape character, which means that it has a special meaning in strings, its exact behaviour is decided by the character following it..

You want a literal backslash in your string, so to do that you need to escape the backslash, this is done by placing two backslashes every time you want one to be in the string.

```lua
print("\\") --> "\"
```

The reasons that your string is breaking there is that `\"` tells lua to put the `"` character in the string and not to close that string. So things break when lua gets to the end of the line thinking that you wanted to escape the closing quote and thus not finding a quote mark that closes the string before the end of the line. So it ends up with a partial string with your would-be closing quote and your closing bracket.
